### PR TITLE
refactor: move internal debugging functions to dedicated namespace

### PR DESCRIPTION
**Pull request by mlutze**
related to #4770

### CODE REVIEW

**Comment by magnus-madsen** [comment]
I am not sure agree with these changes... we can discuss :)

**Comment by magnus-madsen** [comment]
Inline `use`s

**Comment by magnus-madsen** [comment]
Build failed.

File: main/test/ca/uwaterloo/flix/library/TestDebug.flix [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+    def thunkFunction(x: a): String -> a = _ -> x
+
+    @test
+    def testFunctionStringify03(): Bool = {
+        let s = stringify((thunkFunction(42), thunkFunction('a')));
+        s == "(Obj -> Int32, Obj -> Char)"
+    }
+
+    // @test
```
Comment (magnus-madsen) [inline main/test/ca/uwaterloo/flix/library/TestDebug.flix:194]: Why commented?

Comment (mlutze) [reply main/test/ca/uwaterloo/flix/library/TestDebug.flix:194]: Good question

**Pull request merged**